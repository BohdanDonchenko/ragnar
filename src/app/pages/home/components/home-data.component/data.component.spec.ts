import { TestBed, async } from '@angular/core/testing';
import { DataComponent } from './data.component';

describe('DataComponent', () => {
  describe('Snapshot', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        declarations: [DataComponent],
      }).compileComponents();
    }));

    test('check snapshot', () => {
      const fixture = TestBed.createComponent(DataComponent);

      expect(fixture).toMatchSnapshot();
    });
  });

  describe('Functionality', () => {
    test('update', () => {
      const sut = new DataComponent();
      sut.onUpdated.emit = jest.fn();
      sut.click();
      expect(sut.onUpdated.emit).toBeCalledTimes(1);
    });
  });
});
