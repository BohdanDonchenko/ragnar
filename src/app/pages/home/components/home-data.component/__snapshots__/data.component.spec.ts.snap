// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataComponent Snapshot check snapshot 1`] = `
<data-component
  data="0"
  onUpdated={[Function EventEmitter_]}
  title=""
>
  <div
    class="home-data"
  >
    <button>
      
    </button>
    <h4>
      Data: 
    </h4>
  </div>
</data-component>
`;
